audit/hashstructure_test.go

```diff
@@ -392,3 +392,53 @@ func TestHashWalker_TimeStructs(t *testing.T) {
 		}
 	}
 }
+
+// Test_hashWalker_Primitive_TypeDefinition is to avoid future regression in the hashWalker.Primitive method.
+func Test_hashWalker_Primitive_TypeDefinition(t *testing.T) {
+	callback := func(input string) string { return "***" }
+	t.Run("simple-map", func(t *testing.T) {
+		m := map[string]interface{}{"key": "value1"}
+		err := hashMap(callback, m, nil)
+		require.NoError(t, err)
+		require.Equal(t, "***", m["key"])
+	})
+	t.Run("map-with-anon-struct", func(t *testing.T) {
+		m := map[string]interface{}{
+			"key": struct{ Value string }{Value: "value1"},
+		}
+		err := hashMap(callback, m, nil)
+		require.NoError(t, err)
+		require.Equal(t, "value1", m["key"].(struct{ Value string }).Value)
```

Comment:
Are the test cases where we have expected values that are not `***` from where we'd panic before but now return nil? I guess the implication is that these wouldn't be hmac'd in the audit logs?

I guess it would become a concern more generally that you wouldn't want to return a secret in one of those unsettable types, since it wouldn't be hmac'd in the audit log. That said, a panic is also not great 😅

Reply:
Yeah, that's right. These are the test cases that would've panic'd, but now just returns early instead of applying the hmac callback func.